codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -1173,6 +1180,8 @@ private static boolean loadDataMultipartOptimized(ByteBuf undecodedChunk, String
                 undecodedChunk.writerIndex(startReaderIndex);
                 return false;
             } else if (posDelimiter > 0) {
+                // posDelimiter is not from startReaderIndex but from startReaderIndex + lastPosition
+                posDelimiter += lastPosition;
```

Comment:
The `else` branch can be unwrapped since the `if` branch above never completes normally.

Reply:
I mean that

```
if … {
    return …;
} else if … {
    return …;
}
return …;
```

Can be written as

```
if … {
    return …;
}
if … {
    return …;
}
return …;
```

Intellij likes to point this out.